rclrs/build.rs

```diff
@@ -29,7 +63,9 @@ fn main() {
 
     if let Ok(ament_prefix_paths) = ament_prefix_var {
         for ament_prefix_path in ament_prefix_paths.split(':') {
-            builder = builder.clang_arg(format!("-I{}/include", ament_prefix_path));
+            if let Some(clang_args) = generate_clang_include_arg(ament_prefix_path) {
```

Comment:
Could the `generate_clang_include_arg()` function be simply replaced by
```
builder = builder.clang_arg(format!("-I{}/include", ament_prefix_path));
builder = builder.clang_arg(format!("-I{}/include/rcl", ament_prefix_path));
```
? After all it's only rcl that has this problem.

Reply:
I thought that since `rcl` is adopting this change due to concerns about overlays, other packages would soon follow the new format. And I might be misremembering, but I thought a few already had... I could be wrong though.

Reply:
Please revert to the simpler format. Besides the extra lines of code, I can see the following issues with the "general" solution:
- You take the directory name of the prefix path, but for --merge-install, that would be e.g. "rolling", not the package name, right?
- It makes the whole include system looser, i.e. if a package ever decides to both provide `<package_name>/foo.h` and `foo.h`, you may get `<package_name>/foo.h`  even though you asked for `foo.h`.

Reply:
So the trouble I'm running into, is that I have to keep extending the number of "special" include statements to get things to compile: In addition to `rcl` needing that special include pattern, I've also had to do `rcutils`, `rmw`, `rcl_yaml_param_parser`, and `rosidl_runtime_c` (so far, I'm still going through them), which was why I did that generic function.

If the generic function as it is written currently is not correct, is there a better way to write it?


> * You take the directory name of the prefix path, but for --merge-install, that would be e.g. "rolling", not the package name, right?

I don't _think_ so? But I use `--symlink-install`, so I'm not sure...

Reply:
> I don't think so? But I use --symlink-install, so I'm not sure...

`AMENT_PREFIX_PATH` is `/opt/ros/rolling` on a merged installation (like the one in my environment), so `project_name` will be `rolling`. So I don't think this approach will work, but you can try it out if you're not convinced (I don't have a Rolling installation ready at the moment).

Annoying that there are more packages than `rcl` with this issue. I guess we have to add all of them, at least I can't think of a cleaner solution right now.